lib/praxis/response_definition.rb

```diff
@@ -287,10 +287,17 @@ def validate_body!(response)
     def validate_parts!(response)
       return unless parts
 
-      response.parts.each do |name, part|
-        parts.validate(part)
+      case response.body
+      when Praxis::Types::MultipartArray
+        response.body.each do |part|
```

Comment:
This is slightly different. `response.body.validate` is covered by the general response body validation. This is for validating the explicit response part definition for things like headers and general content type.